src/subgraphs/getData.ts

```diff
@@ -83,31 +86,50 @@ export class DataRecoveryGraph extends AbstractGraph<DataRecoveryState> {
         value: (x: string, y?: string) => (y ? y : x),
         default: () => '',
       },
+      status: {
+        value: (x: NodeStatus | null, y?: NodeStatus | null) => (y ? y : x),
+        default: () => null,
+      },
     };
     super(graphState);
     this.functions = functions;
     this.database = database;
     this.companyName = companyName;
+    this.outputHandler = outputHandler;
+  }
+
+  private async handleStatusUpdate(status: NodeStatus) {
+    if (status) {
+      if ('error' in status) {
+        this.outputHandler('error', status.error!.message);
+      } else if (status.type = 'data') {
+        this.outputHandler('status', status.data!.message);
+      }
+    }
   }
 
   private async fetchSchemaNode(state: DataRecoveryState): Promise<DataRecoveryState> {
-    const { sessionToken, schema } = await fetchSchema(this.companyName, this.database);
-    return { ...state, sessionToken, schema };
+    const { sessionToken, schema, status } = await fetchSchema(this.companyName, this.database);
+    this.handleStatusUpdate(status);
+    return { ...state, sessionToken, schema, status };
   }
 
   private async checkUpdateSemanticLayer(state: DataRecoveryState): Promise<DataRecoveryState> {
-    const {needsSemanticUpdate, semanticTask} = await checkUpdateSemanticLayer(state.task, this.companyName);
-    return { ...state, needsSemanticUpdate, semanticTask };
+    const {needsSemanticUpdate, semanticTask, status} = await checkUpdateSemanticLayer(state.task, this.companyName);
+    this.handleStatusUpdate(status);
+    return { ...state, needsSemanticUpdate, semanticTask, status };
   }
 
   private async handleEditCubeGraph(state: DataRecoveryState): Promise<DataRecoveryState> {
-    const { schema } = await handleEditCubeGraph(state.semanticTask, state.sessionToken, this.functions, this.database, this.companyName);
-    return { ...state, schema };
+    const { schema, status } = await handleEditCubeGraph(state.semanticTask, state.sessionToken, this.functions, this.database, this.companyName);
```

Comment:
Same as metabase
If we use this in a handler in the router for example, the status is going to be confusing no?